tasks/ui_framework.js

```diff
@@ -6,6 +6,37 @@ import debounce from 'lodash/function/debounce';
 const platform = require('os').platform();
 
 module.exports = function (grunt) {
+  grunt.registerTask('uiFramework:build', function () {
+    const done = this.async();
+
+    const serverCmd = {
+      cmd: /^win/.test(platform) ? '.\\node_modules\\.bin\\webpack.cmd' : './node_modules/.bin/webpack',
+      args: [
+        '-p',
+        '--config=ui_framework/doc_site/webpack.config.js',
+      ],
+      opts: { stdio: 'inherit' }
+    };
+
+    const uiFrameworkServerBuild = new Promise((resolve, reject) => {
```

Comment:
Maybe I'm misunderstanding, but I just tried returning the promise and removed `done`, and the process just hangs. So I think we need to keep it as-is?